src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/PdfService.kt

```diff
@@ -8,48 +10,96 @@ import org.springframework.stereotype.Service
+    prisonNumber: String,
+    data: HashMap<String, Any>,
+  ) {
+    when (documentSubjectType) {
+      DocumentSubjectType.OFFENDER_ELIGIBLE_FORM,
+      DocumentSubjectType.OFFENDER_ADDRESS_CHECKS_INFORMATION_FORM,
+      DocumentSubjectType.OFFENDER_ADDRESS_CHECKS_FORM,
+      DocumentSubjectType.OFFENDER_OPT_OUT_FORM,
+      DocumentSubjectType.OFFENDER_NOT_ELIGIBLE_FORM,
+      DocumentSubjectType.OFFENDER_ADDRESS_UNSUITABLE_FORM,
+      DocumentSubjectType.OFFENDER_POSTPONED_FORM,
+      DocumentSubjectType.OFFENDER_NOT_ENOUGH_TIME_FORM,
+      DocumentSubjectType.OFFENDER_APPROVED_FORM,
+      DocumentSubjectType.OFFENDER_AGENCY_NOTIFICATION_FORM,
+      DocumentSubjectType.OFFENDER_CANCEL_AGENCY_NOTIFICATION_FORM,
+      DocumentSubjectType.OFFENDER_REFUSED_FORM,
+      DocumentSubjectType.OFFENDER_NOT_SUITABLE_FORM,
+      -> {
+        val currentAssessment = assessmentService.getCurrentAssessmentSummary(prisonNumber)
+        data["currentAssessment"] = currentAssessment
+      }
+    }
+  }
+
+  private fun createHtmlContent(templatePathAndFile: String, details: Map<String, Any>): String {
+    log.debug("Creating html content using $templatePathAndFile")
+    val context = Context()
+    context.setVariables(details)
+    return thymeleafEngine.process(templatePathAndFile, context)
+  }
+
+  private fun createPdf(htmlContent: String, documentSubjectType: DocumentSubjectType): ByteArray? {
     val headers = HttpHeaders()
     headers.contentType = MediaType.MULTIPART_FORM_DATA
 
-    val body = LinkedMultiValueMap<String, Any>()
-    body.add(
-      "files",
-      HttpEntity(
-        htmlContent.toByteArray(StandardCharsets.UTF_8),
-        HttpHeaders().apply {
-          contentType = MediaType.TEXT_HTML
-          setContentDispositionFormData("files", "index.html")
-        },
-      ),
+    val documentData = HttpEntity(
+      htmlContent.toByteArray(StandardCharsets.UTF_8),
+      HttpHeaders().apply {
+        contentType = MediaType.TEXT_HTML
+        setContentDispositionFormData("files", documentSubjectType.name.lowercase() + ".pdf")
+      },
     )
-    body.add("paperWidth", "8.27")
-    body.add("paperHeight", "11.69")
-    body.add("marginTop", "1")
-    body.add("marginBottom", "1")
-    body.add("marginLeft", "1")
-    body.add("marginRight", "1")
 
-    val requestEntity = HttpEntity(body, headers)
+    val body = LinkedMultiValueMap<String, Any>()
+    body.add("files", documentData)
+    body.add("paperWidth", pdfConfigProperties.paperWidth)
+    body.add("paperHeight", pdfConfigProperties.paperHeight)
+    body.add("marginTop", pdfConfigProperties.marginTop)
+    body.add("marginBottom", pdfConfigProperties.marginBottom)
+    body.add("marginLeft", pdfConfigProperties.marginLeft)
+    body.add("marginRight", pdfConfigProperties.marginRight)
 
-    return gotenbergApiClient.convertHtmlToPdf(requestEntity)
+    return gotenbergApiClient.requestPdf(HttpEntity(body, headers))
   }
 }
```

Comment:
@andrewrlee & @nishanthmahasamudram  Should this code be encapsulated in the GotenbergApiClient? This is about how we request the data from gotenberg api, if we do this we can test it separately if needed!